templates/custom_search.pages.inc

```diff
@@ -29,7 +29,7 @@ function custom_search_preprocess_search_results(&$variables) {
         $keys = backdrop_substr($keys, 0, strpos($keys, 'type:') - 1);
       }
       // Get Custom Search authorized types.
-      $searchable_node_types = array();
+      $searchable_node_types = $defined_types = array();
       if ($defined_types = $config->get('current_node_types')) {
```

Comment:
I haven't tested, but this fix doesn't seem right. the `if` statement has an assignment `=` rather than a comparison `==`, so `$defined_types` will be reassigned within the `if` statement, effectively undoing line 32 and throwing the warning again. Without having looked at the whole code or tested, I think the problem is that  in some instances `$config->get('current_node_types'))` will return NULL, and the `array_filter` will complain, so, the solution is to check for that AFTER the assignment.

Reply:
OK Scratch that! Too early for me. Yes, this will fix the problem as the `if` statement will evaluate as FALSE if the variable is NULL.

I'll test next.

Reply:
This works for php 7 but not for php 8. I've working with herbdool to resolve.

Reply:
I've updated the PR.